### PR TITLE
Critical amendment to GetWorkClient

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Addresses #1213

- Solution submit is immediate upon arrival and is no influenced by --farm-recheck value
- Submission of hashrate is conditional
- Renamed m_report_stratum_hashrate to m_report_hashrate as it's valid for both modes

Now even sligtly higher --farm-recheck values may be applied to mitigate overhead.

### CODE REVIEW

File: libpoolprotocols/getwork/EthGetworkClient.h [hunk 1]
```diff
@@ -13,7 +13,7 @@ using namespace eth;
 class EthGetworkClient : public PoolClient, Worker
 {
 public:
-	EthGetworkClient(unsigned const & farmRecheckPeriod);
+	EthGetworkClient(unsigned const & farmRecheckPeriod, bool const & submitHashrate);
```
Comment (chfast) [inline libpoolprotocols/getwork/EthGetworkClient.h]: This is wrong. Pass both arguments by value.

File: libpoolprotocols/getwork/EthGetworkClient.cpp [hunk 2]
```diff
@@ -53,13 +51,35 @@ void EthGetworkClient::disconnect()
...
 	m_currentHashrateToSubmit = rate;
+
 }
 
 void EthGetworkClient::submitSolution(Solution solution)
 {
-	// Store the solution in temp var. Will be handled in workLoop
-	m_solutionToSubmit = solution;
+	// Immediately send found solution without wait for loop
+	if (m_connected || m_justConnected) {
```
Comment (chfast) [inline libpoolprotocols/getwork/EthGetworkClient.cpp]: What is the `m_justConnected` flag for?

Comment (chfast) [inline libpoolprotocols/getwork/EthGetworkClient.cpp]: What is the `m_justConnected` flag for?

**Comment by chfast** [comment]
Also add CHANGELOG entry.

File: libpoolprotocols/getwork/EthGetworkClient.cpp (hunk 2 above)
Comment (AndreaLanfranchi) [reply libpoolprotocols/getwork/EthGetworkClient.cpp]: Really do not know. I only replicated what i believe is the "connected" condition imposed in workloop. I did not write this class.

Comment (AndreaLanfranchi) [reply libpoolprotocols/getwork/EthGetworkClient.cpp]: Really do not know. I only replicated what i believe is the "connected" condition imposed in workloop. I did not write this class.

**Pull request merged**